reps/2023-12-14-virtual-cluster.md

```diff
@@ -0,0 +1,452 @@
+# Virtual Cluster
+
+## Summary
+
+In addition to the existing [physical cluster](https://docs.ray.io/en/releases-2.9.0/cluster/getting-started.html) and physical node concepts, we propose to add new virtual cluster and virtual node concepts. A virtual node is a partition of a single physical node and it has resources and node labels just like the physical node. A virtual cluster made up of virtual nodes is a partition of the physical cluster and can be dynamically scaled at runtime. A physical cluster can be partitioned into multiple virtual clusters and each virtual cluster runs a single Ray job. Through this way, multiple jobs can share a physical cluster with logical resources isolation. Virtual cluster is a fundamental building block for multi-tenant Ray.
```

Comment:
Do we plan to support different "environments" for different virtual clusters within the same physical cluster? E.g. different Python versions?

Reply:
This is out of scope for virtual cluster but potentially in scope for multi-tenant Ray.